### DISCUSSION TITLE
table display options

### DISCUSSION THREAD

**Discussion by proxy318** [Budibase/budibase#9638] [Ideas]
When you use the table component to display data, if the data is to wide for a cell, it is truncated with ellipses. It would nice if you could either expand the column to show the complete data, or if the full data would appear in a tool-tip or something.

**Comment by andz-bb** [comment]
hey @proxy318 - this is a great idea, thanks for sharing!

If you need to see the full value in the meantime, you can manually adjust the width of table columns by clicking _Configure columns_ in the Table component settings. Then if you click Add all rows there will be a settings cog beside each column where you can override the automatic width setting.

![column width adjustment](https://user-images.githubusercontent.com/110921612/218065310-d85f79b0-7dad-45c0-b021-bf9e6733d24c.gif)

This works well when short strings (like phone numbers in my example) are being truncated but it's not ideal for larger data so we will definitely keep your suggestion in mind for future development 👍

**Reply by proxy318** [reply]
Thanks, I was aware of the column customization, but as you said, this doesn't work for longer data. I'm using the table to display, among other things, email addresses, and with the other data I have, the only way I can get the email addresses to fit is to truncate some of my other columns.

**Reply by Wulkan93** [reply]
Has this been fixed yet? I have the same problem as proxy318. Have one column that explains some routines to be done when testing objects connected to a project. Would be great if the text was able to be shown in its entirety instead of truncated.

**Reply by joebudi** [reply]
Have you tried changing the height of the row, or the width of the column (column settings)?

**Reply by Wulkan93** [reply]
Thank you for your reply @joebudi

I need the cell size to be able to start out small so it doesn't take too much space. But also need the ability to show the content, for example by clicking on the cell-value then either the cell will expand or a new window will pop-up, not sure what would be best.. Resizing the width of the column is not a scalable solution since the entirety of the cell value in most cases will be quite long and not enough for just one row.

See below screenshot to get an idea of what I mean:
![image](https://github.com/user-attachments/assets/7e49f4e4-48f6-4754-bb8b-ec4eae12cdc4)

The concerned cell is the one highlighted in the middle-table under the column Routine.

**Reply by Wulkan93** [reply]
Perhaps if it was possible to link a pdf or some other sort of document directly to the cell and then that document opens when the cell is clicked?

**Reply by joebudi** [reply]
You could add an `on row click` action to the table to open a modal/side panel with the cell's details. Or, if you make the table editable, and the field type is 'long form text', the text will open within a popover when clicked.

**Reply by Wulkan93** [reply]
@joebudi Opening up a modal to show the content sounds like it could work. However, I can't really find how to enlarge the cell with the content enough. I've added a table in the modal, then selected so that only the column Routine is showing and adjusted the width and height (height was maximum setting to put "Large"). But I'd like to have like a full window displaying the content and then the option to scroll. Maybe I took a wrong turn somewhere?

See picture of current display.
![image](https://github.com/user-attachments/assets/f0d2ae44-1734-4fd4-8c0b-336a75b3a716)

**Reply by joebudi** [reply]
You could show the cell value in a different way, using a different component - for example, the long-form text component. 

The following doc will help. It's a side panel, but the experience is roughly the same.
https://docs.budibase.com/docs/side-panel

**Reply by Wulkan93** [reply]
@joebudi Thank you. Using a Long Form Field inside a Modal that opens up through an on row click-action worked perfectly.
